internal/service/kubernetes/node_group.go

```diff
@@ -2,470 +2,674 @@ package kubernetes
-							Description: "Kubelet argument key.",
-							Type:        schema.TypeString,
-							Required:    true,
-							ValidateDiagFunc: validation.ToDiagFunc(
-								validation.StringMatch(regexp.MustCompile("^[a-zA-Z0-9-]+$"), ""),
-							),
+			"utility_network_access": schema.BoolAttribute{
+				MarkdownDescription: "If set to false, nodes in this group will not have access to utility network.",
+				Computed:            true,
+				Optional:            true,
+				Default:             booldefault.StaticBool(true),
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"custom_plan": schema.ListNestedBlock{
+				MarkdownDescription: "Resource properties for custom plan",
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.RequiresReplace(),
+					planmodifierutil.CustomPlanPlanModifier(),
+				},
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+				},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"cores": schema.Int64Attribute{
+							MarkdownDescription: "The number of CPU cores dedicated to individual node group nodes when using custom plan",
+							Required:            true,
+							PlanModifiers: []planmodifier.Int64{
+								int64planmodifier.RequiresReplace(),
+							},
+							Validators: []validator.Int64{
+								int64validator.Between(1, 20),
+							},
+						},
+						"memory": schema.Int64Attribute{
+							MarkdownDescription: "The amount of memory in megabytes to assign to individual node group node when using custom plan. Value needs to be divisible by 1024.",
+							Required:            true,
+							PlanModifiers: []planmodifier.Int64{
+								int64planmodifier.RequiresReplace(),
+							},
+							Validators: []validator.Int64{
+								int64validator.Between(2048, 131072),
+								validatorutil.DivisibleBy(1024),
+							},
+						},
+						"storage_size": schema.Int64Attribute{
+							MarkdownDescription: "The size of the storage device in gigabytes.",
+							Required:            true,
+							PlanModifiers: []planmodifier.Int64{
+								int64planmodifier.RequiresReplace(),
+							},
+							Validators: []validator.Int64{
+								int64validator.Between(25, 1024),
+							},
 						},
-						"value": {
-							Description:      "Kubelet argument value.",
-							Type:             schema.TypeString,
-							Required:         true,
-							ValidateDiagFunc: validation.ToDiagFunc(validation.StringLenBetween(0, 255)),
+						"storage_tier": schema.StringAttribute{
+							MarkdownDescription: fmt.Sprintf("The storage tier to use. Defaults to %s", upcloud.KubernetesStorageTierMaxIOPS),
+							Optional:            true,
+							Computed:            true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+							Validators: []validator.String{
+								stringvalidator.OneOf(
```

Comment:
Does this support standard yet, or is it waiting for update on the service side like the developer plans?

Reply:
Added alongside upcloud-go-api v8.7.1